restcomm/restcomm.http/src/main/java/org/restcomm/connect/http/IncomingPhoneNumbersEndpoint.java

```diff
@@ -654,4 +650,108 @@ public Response deleteIncomingPhoneNumber(final String accountSid, final String
                 incomingPhoneNumber.isFaxCapable(),
                 false);
     }
+
+    /**
+     * @param targetAccountSid
+     * @param data
+     * @param responseType
+     * @return
+     */
+    protected Response migrateIncomingPhoneNumbers(String targetAccountSid, MultivaluedMap<String, String> data, MediaType responseType) {
+        Account effectiveAccount = userIdentityContext.getEffectiveAccount();
+        secure(effectiveAccount, "RestComm:Modify:IncomingPhoneNumbers");
+        allowOnlySuperAdmin();
+        try{
+            Account targetAccount = accountsDao.getAccount(targetAccountSid);
+            // this is to avoid if mistakenly provided super admin account as targetAccountSid
+            // if this check is not in place and someone mistakenly provided super admin
+            // then all accounts and sub account in platform will be impacted
+            if(targetAccount.getParentSid() == null){
```

Comment:
do we need to check if targetAccount is null, or can we assume this will alwasy be non Null?

Reply:
handled